python/lsst/pipe/tasks/postprocess.py

```diff
@@ -1088,6 +1093,187 @@ def run(self, handle, funcs=None, dataId=None, band=None, **kwargs):
         return pipeBase.Struct(outputCatalog=tbl)
 
 
+class ComputeColumnsAction(ConfigurableAction):
+    """An action that computes multiple vectors from an input.
+
+    This class is meant to be compatible with analysis_tools'
+    AnalysisAction class, which cannot be a dependency of pipe_tasks."""
+
+    def getInputSchema(self) -> dict[str, type[NDArray]]:
+        """Return the required inputs for this action.
+
+        This function is meant to be compatible with
+        """
+        raise NotImplementedError("This method must be overloaded in subclasses")
+
+    def __call__(self, table: astropy.table.Table) -> dict[str, NDArray]:
+        """This method must return a dict of computed columns."""
+        raise NotImplementedError("This method must be overloaded in subclasses")
+
+
+class ExtendednessColumnActionBase(ComputeColumnsAction):
+    bands = pexConfig.ListField[str](
+        doc="The bands to make single-band outputs for.",
+        default=["u", "g", "r", "i", "z", "y"]
+    )
+    bands_combined = pexConfig.DictField[str, str](
+        doc="Multiband classification column specialization. Keys specify the"
+            " name of the column and values are a comma-separated list of"
+            " bands, all of which must be contained in the bands listed.",
+        default={"griz": "g,r,i,z"},
+        itemCheck=lambda x: (len(y := x.split(",")) > 1) & (len(set(y)) == len(y)),
+    )
+    model_column_flux = pexConfig.Field[str](
+        doc="The model flux column to use for computing the difference to"
+            " to the S/N flux. Must contain the {band} and {model} templates.",
+        default="{band}_{model}Flux",
```

Comment:
Could this be `{band}_sersicFlux` (and `{band}_sersicFluxErr`) instead of needing a separate parameter for the model?